### PR TITLE
Add i18n lint step using WP-CLI

### LINKED ISSUE #6940: Internationalization: Check correct configuration in CI
## Task Description

This is a follow-up to #6910.

It would be nice if we could detect correct configuration for string extraction on CI, for example by simply running WP-CLI  and see if there are loads of strings being found or not.

This can run on a schedule (e.g. daily) as it would be wasteful to run on every PR.

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Adds a i18n lint step using WP-CLI.

Also massively improves translatable strings based on localization team's feedback.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6940

### CODE REVIEW

**Comment by swissspidy** [comment]
Here's the check in action: https://github.com/google/web-stories-wp/pull/7216/checks?check_run_id=2352015640#step:15:18

File: .github/workflows/lint-i18n.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+      # TODO: Switch to WP-CLI stable (wp-cli.phar) once 2.5.0 is out.
+      - name: Install WP-CLI
+        run: |
+          curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar
+          mv wp-cli-nightly.phar wp-cli.phar
+          chmod +x wp-cli.phar
+          mkdir -p bin
+          mv wp-cli.phar bin/wp
+          echo "${PWD}/bin" >> $GITHUB_PATH
```
Comment (spacedmonkey) [inline .github/workflows/lint-i18n.yml:45]: I am not sure I love how WP CLI is installed here. Can we either find a container that already has WP CLI installed or use composer.

File: .github/workflows/lint-i18n.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
+name: Lint I18N
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 12 * * 1-5'
```
Comment (spacedmonkey) [inline .github/workflows/lint-i18n.yml]: Can you explain cron / scheduled and why it is needed?

**Comment by swissspidy** [comment]
@spacedmonkey @choumx would appreciate your review here

File: .github/workflows/lint-i18n.yml (hunk 2 above)
Comment (dreamofabear) [inline .github/workflows/lint-i18n.yml]: Nit: Add a comment describing the cron timing here.

File: .github/workflows/lint-i18n.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # TODO: Switch to WP-CLI stable (wp-cli.phar) once 2.5.0 is out.
+      - name: Install WP-CLI
+        run: |
+          curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar
```
Comment (dreamofabear) [inline .github/workflows/lint-i18n.yml:40]: Running nightly might be brittle. Is the intent to pick up the false positive bug fix sooner?

**Review by dreamofabear — APPROVED** [review]

File: .github/workflows/lint-i18n.yml (hunk 3 above)
Comment (swissspidy) [reply .github/workflows/lint-i18n.yml:40]: As a maintainer of WP-CLI I can assure you the current nightly is quite stable :-)

The nightly contains some important bugfixes for newer PHP versions.

The next stable is already around the corner, so we can switch soon.

File: .github/workflows/lint-i18n.yml (hunk 1 above)
Comment (swissspidy) [reply .github/workflows/lint-i18n.yml:45]: This is the recommended way to install WP-CLI.

Using Docker or Composer just for this seems way overkill.

File: .github/workflows/lint-i18n.yml (hunk 3 above)
Comment (spacedmonkey) [reply .github/workflows/lint-i18n.yml:40]: > As a maintainer of WP-CLI I can assure you the current nightly is quite stable :-)

You may not always be on the project :P

File: .github/workflows/lint-i18n.yml (hunk 2 above)
Comment (swissspidy) [reply .github/workflows/lint-i18n.yml]: Will add a comment. Basically, no need to run this for every PR/push, so cron saves some resources here.

**Review by spacedmonkey — CHANGES_REQUESTED** [review]
This needs a rebase.

**Pull request merged**